src/number_theory/pythagorean_triples.lean

```diff
@@ -534,6 +534,54 @@ begin
     { split, { ring }, rw int.gcd_comm, exact coprime_pow_two_sub_mul co pp } }
 end
 
+/-- by assuming `x` is odd and `z` is positive we get a slightly more precise classification of
+the pythagorean triple `x ^ 2 + y ^ 2 = z ^ 2`-/
+theorem coprime_classification' {x y z : ℤ} (h : pythagorean_triple x y z)
+  (h_coprime : int.gcd x y = 1) (h_parity : x % 2 = 1) (h_pos : 0 < z) :
+  ∃ m n,  x = m ^ 2 - n ^ 2
+        ∧ y = 2 * m * n
+        ∧ z = m ^ 2 + n ^ 2
+        ∧ int.gcd m n = 1
+        ∧ ((m % 2 = 0 ∧ n % 2 = 1) ∨ (m % 2 = 1 ∧ n % 2 = 0))
+        ∧ 0 ≤ m :=
+begin
+  obtain ⟨m, n, ht1, ht2, ht3, ht4⟩ :=
+    pythagorean_triple.coprime_classification.mp (and.intro h h_coprime),
+  by_cases hm_sign : 0 ≤ m,
+  { use [m, n],
+    cases ht1 with h_odd h_even,
+    { apply and.intro h_odd.1,
+      apply and.intro h_odd.2,
+      cases ht2 with h_pos h_neg,
+      { apply and.intro h_pos (and.intro ht3 (and.intro ht4 hm_sign)) },
+      { exfalso, revert h_pos, rw h_neg,
+        exact imp_false.mpr (not_lt.mpr (neg_nonpos.mpr (add_nonneg (pow_two_nonneg m)
+          (pow_two_nonneg n)))) } },
+    exfalso,
+    replace h_even : 2 ∣ x, { apply dvd.intro (m * n), rw [h_even.1, mul_assoc] },
+    replace h_even : x % 2 = 0, { exact int.mod_eq_zero_of_dvd h_even },
+    rw h_parity at h_even, revert h_even, norm_num },
+  { use [-m, -n],
+    cases ht1 with h_odd h_even,
+    { rw [neg_square m],
+      rw [neg_square n],
+      apply and.intro h_odd.1,
+      split, { rw h_odd.2, ring },
+      cases ht2 with h_pos h_neg,
+      { apply and.intro h_pos,
+        split,
+        { delta int.gcd, rw [int.nat_abs_neg, int.nat_abs_neg], exact ht3 },
+        { rw [int.neg_mod_two, int.neg_mod_two],
+          apply and.intro ht4, linarith } },
+      { exfalso, revert h_pos, rw h_neg,
+        exact imp_false.mpr (not_lt.mpr (neg_nonpos.mpr (add_nonneg (pow_two_nonneg m)
+          (pow_two_nonneg n)))) } },
+    exfalso,
+    replace h_even : 2 ∣ x, { apply dvd.intro (m * n), rw [h_even.1, mul_assoc] },
+    replace h_even : x % 2 = 0, { exact int.mod_eq_zero_of_dvd h_even },
+    rw h_parity at h_even, revert h_even, norm_num }
```

Comment:
```suggestion
    rcases h_even with ⟨rfl, -⟩,
    rw [mul_assoc, int.mul_mod_right] at h_parity,
    exact zero_ne_one h_parity }
```

Reply:
The rcases rfl trick removes `x` completely making things a bit simpler.

Reply:
@paulvanwamelen What do you think of this suggestion?

Reply:
This was adopted verbatim.

Reply:
Aah, for some reason github is not realising this...

Reply:
And I also don't see the adoption when looking at the diff. Maybe something went wrong?